### PR TITLE
Fix issue with passwords when passwords is provided by dict and by file

### PR DESCRIPTION
**Pull request by FloLaco**
Issue: #809 

Fix the issue about `dict.update()` which return `None`. It's a bad usage of this method.

### CODE REVIEW

**Comment by Shrews** [comment]
That certainly does look like problematic code. Could you add a test that validates the fix?

**Comment by FloLaco** [comment]
@Shrews I've added some tests

**Comment by Shrews** [comment]
recheck

File: ansible_runner/config/_base.py [hunk 1]
```diff
@@ -150,16 +150,24 @@ def _prepare_env(self, runner_mode='pexpect'):
         if self.runner_mode == 'pexpect':
             try:
                 if self.passwords and isinstance(self.passwords, dict):
-                    self.passwords = self.passwords.update(self.loader.load_file('env/passwords', Mapping))
+                    # Load password file and then the passwords dict argument for precedence
```
Comment (Shrews) [inline ansible_runner/config/_base.py:153]: What's the reason for the change in precedence?

Comment (FloLaco) [reply ansible_runner/config/_base.py:153]: Because it's useless to provide password as dictionnary if a file is already present if we don't not modify precedence. 
Argument provided by "user" in python code should always be more important as config file.

Comment (AlanCoding) [reply ansible_runner/config/_base.py:153]: @Shrews made a good point to me

```python
self.settings.update(self.loader.load_file('env/settings', Mapping))
```

differs from your precedence rules for passwords here.

I agree with you sentiment

> Argument provided by "user" in python code should always be more important as config file.

But that would apply to settings, just the same that it applies to passwords.

Comment (FloLaco) [reply ansible_runner/config/_base.py:153]: @AlanCoding you're right.
I didn't change settings because I don't not use it and the initial fix was not intend to change settings. I just changed the obvious bug about settings.
If you insist I change change precedence with settings to, but I just need to know how to test it.

Comment (Shrews) [reply ansible_runner/config/_base.py:153]: IMO, all entries in `env` should have the same precedence. I feel like this PR should just fix the current bug. Changing precedence for file vs. args should likely be a separate PR and make sure that precedence is treated the same way across all possible input files.

**Comment by samdoran** [comment]
Superseded by #899
